reign.py

```diff
@@ -116,15 +121,27 @@ def battle(self):
         """
 
         # Choose ∑
-        empire = random.choice(self.obj.Empire.values.tolist())
-        empire_neighbours = self.obj.query(f'Empire == "{empire}"').iloc[0].empire_neighbours
+
+        empire_list = self.obj.Empire.values.tolist()
+        un_empire, empire_weight = np.unique(empire_list, return_counts=True)
+        probability_ratio = max(empire_weight)/empire_weight
+
+        if len(un_empire) < self.threshold and any(probability_ratio > self.low_b):
+            empire_weight[probability_ratio > self.low_b] = max(empire_weight)/self.low_b
+        else:
+            pass
+
+        empire=random.choices(un_empire, empire_weight)
```

Comment:
Should `empire` be called `empires`?
Use spaces `empire = random.choices(un_empire, empire_weight)`

Reply:
Otherwise, can we define `empire = random.choices(un_empire, empire_weight)[0]`